silverback/application.py

```diff
@@ -1,17 +1,42 @@
 import atexit
+from dataclasses import dataclass
 from datetime import timedelta
 from typing import Callable, Dict, Optional, Union
 
 from ape.api.networks import LOCAL_NETWORK_NAME
 from ape.contracts import ContractEvent, ContractInstance
 from ape.logging import logger
 from ape.managers.chain import BlockContainer
-from ape.types import AddressType
 from ape.utils import ManagerAccessMixin
 from taskiq import AsyncTaskiqDecoratedTask, TaskiqEvents
 
-from .exceptions import DuplicateHandlerError, InvalidContainerTypeError
+from .exceptions import InvalidContainerTypeError
 from .settings import Settings
+from .types import TaskType
+
+
+@dataclass
+class Task:
+    container: Union[BlockContainer, ContractEvent, None]
+    handler: AsyncTaskiqDecoratedTask
+
+
+class TaskCollection(dict):
```

Comment:
Also, made this data structure to avoid having to double check if the list was there or not in the dict for each key, open to suggestions otherwise of something more stdlib

Reply:
have you ever tried `defaultdict`? It's pretty cool! It does this I think. Like, you can add `.append` without first having to check if the list exists. It will just work.

https://docs.python.org/3/library/collections.html#collections.defaultdict

Reply:
There we go! Will refactor to use that, as it makes the errors cleaner

Reply:
https://github.com/ApeWorX/silverback/pull/66/commits/e0c3dc7eb210a6aba120b3005d0b4bf6efd807a7